### DISCUSSION TITLE
Training on custom dataset with custom mask for Inpainting

### DISCUSSION THREAD

**Discussion by Akshay-Ijantkar** [open-mmlab/mmediting#1270] [Datasets] [answered]
Hello @lindahua @wwhio @tpoisonooo @hellock @innerlee 
Great work on the paper and even the results look amazing!

Can you please guide me with step by step process on how we need to prepare a dataset for a custom image dataset with a custom mask (if possible)?

How the masks will look like in the sense that whether it will be a colour image on which white patches (the part that will be inpainted from the model) is drawn or it will be a completely black image on which white patches (the part that will be inpainted from the model) is drawn? (in short how to prepare mask and then how to supply it for model training)

I referred to your steps but couldn't completely follow them.
Thank you in advance!

**Comment by wwhio** [comment, marked as answer]
1. Implement your dataset which gives the information of your data. Register it, import it in `mmedit/datasets/__init__.py`, and add it to `__all__`.
2. Implement your pipeline which gives the process according to information provided by dataset. Register it, import it in `mmedit/datasets/pipelines/__init__.py`, and add it to `__all__`.

During implementing your dataset, just give the information:
1. Your dataset should inherit the `ImgInpaintingDataset` class, and implement your `load_annotations` function.
2. You may change the `__init__` function according to your need.
2. In `load_annotations`: package all information you need for one paired data into a `dict` and append these `dict`s to a `list` and return this `list`.
3. Call `load_annotations` in `__init__` like `ImgInpaintingDataset` does.

**Comment by ACALJJ32** [comment]
> 1. Implement your dataset which gives the information of your data. Register it, import it in `mmedit/datasets/__init__.py`, and add it to `__all__`.
> 2. Implement your pipeline which gives the process according to information provided by dataset. Register it, import it in `mmedit/datasets/pipelines/__init__.py`, and add it to `__all__`.
> 
> During implementing your dataset, just give the information:
> 
> 1. Your dataset should inherit the `ImgInpaintingDataset` class, and implement your `load_annotations` function.
> 2. You may change the `__init__` function according to your need.
> 3. In `load_annotations`: package all information you need for one paired data into a `dict` and append these `dict`s to a `list` and return this `list`.
> 4. Call `load_annotations` in `__init__` like `ImgInpaintingDataset` does.

Thank you for this guide,it help me a lot!
